src/region/RegionCreator.tsx

```diff
@@ -207,10 +224,10 @@ export default function RegionCreator({ className, onAbort, onStart, onStop }: P
             document.removeEventListener('mousemove', handleMouseMove);
             document.removeEventListener('mouseup', handleMouseUp);
         };
-    }, [isDrawing]); // eslint-disable-line react-hooks/exhaustive-deps
+    }, [drawingStatus]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
The major downside here is that we will destroy/recreate the render loop and also remove.re-add the handlers when the status changes from `dragging` to `drawing`. Should we only cleanup when in `init` status? Will that lead to bugs or hanging references?

Reply:
I tried to add listeners at the beginning of `dragging` and don't remove them at the end of `dragging` but remove them at the end of `drawing`. Unfortunately it doesn't work. The reason is that every render creates a new `handleMouseUp` function instance. The cleanup callback of `drawing` cannot remove the listener created in `dragging`. We can memo all the functions but I think it's better to do in another PR.